integration_tests/src/main/python/rapids_udf_test.py

```diff
@@ -88,3 +89,24 @@ def evalfn(spark):
         load_java_udf_or_skip_test(spark, 'urlencode', 'com.nvidia.spark.rapids.udf.java.URLEncode')
         return unary_op_df(spark, StringGen('.{0,30}')).selectExpr("urlencode(a)")
     assert_gpu_and_cpu_are_equal_collect(evalfn)
+
+@rapids_udf_example_native
+def test_java_cosine_similarity_reasonable_range(enable_rapids_udf_example_native):
+    def evalfn(spark):
+        class RangeFloatGen(FloatGen):
+            def start(self, rand):
+                def gen_float(): return rand.uniform(-1000.0, 1000.0)
+                self._start(rand, gen_float)
+        load_java_udf_or_skip_test(spark, "cosine_similarity", "com.nvidia.spark.rapids.udf.java.CosineSimilarity")
+        arraygen = ArrayGen(RangeFloatGen(nullable=False), min_length=8, max_length=8)
+        df = binary_op_df(spark, arraygen, length=16)
+        return df.selectExpr("cosine_similarity(a, b)")
+    assert_gpu_and_cpu_are_equal_collect(evalfn)
```

Comment:
Ok so for the CPU test, even though we've registered the `cosine_similarity` udf, the CPU implementation of `cosine_similarity` is picked up in the plan? or is it being skipped, just trying to understand this part.

Reply:
Thanks! Missed that.